File: exercises/exercise-1-testing-java/src/main/java/de/emaarco/example/application/service/bike/SendBikeConfirmationMailService.java
package de.emaarco.example.application.service.bike;

import de.emaarco.example.application.port.inbound.bike.SendBikeConfirmationMailUseCase;
import de.emaarco.example.application.port.outbound.bike.BikeSubscriptionRepository;
import de.emaarco.example.domain.bike.BikeSubscriptionId;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class SendBikeConfirmationMailService implements SendBikeConfirmationMailUseCase {

    private static final Logger log = LoggerFactory.getLogger(SendBikeConfirmationMailService.class);

    private final BikeSubscriptionRepository repository;

    public SendBikeConfirmationMailService(BikeSubscriptionRepository repository) {
        this.repository = repository;
    }

    @Override
    public void sendConfirmationMail(BikeSubscriptionId subscriptionId) {
        var subscription = repository.find(subscriptionId);
        log.info("Sending bike confirmation mail to {} for bike {}",
            subscription.customerEmail().value(), subscription.bikeId().value());
    }

}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/de/emaarco/example/application/service/bike/SendBikeConfirmationMailService.kt
package de.emaarco.example.application.service.bike

import de.emaarco.example.application.port.inbound.bike.SendBikeConfirmationMailUseCase
import de.emaarco.example.application.port.outbound.bike.BikeSubscriptionRepository
import de.emaarco.example.domain.bike.BikeSubscriptionId
import mu.KotlinLogging
import org.springframework.stereotype.Service

@Service
class SendBikeConfirmationMailService(
    private val repository: BikeSubscriptionRepository
) : SendBikeConfirmationMailUseCase {

    private val log = KotlinLogging.logger {}

    override fun sendConfirmationMail(subscriptionId: BikeSubscriptionId) {
        val subscription = repository.find(subscriptionId)
        log.info { "Sending bike confirmation mail to ${subscription.customerEmail.value} for bike ${subscription.bikeId.value}" }
    }
}
